documentapi/src/vespa/documentapi/messagebus/messages/getdocumentreply.h

```diff
@@ -29,28 +31,21 @@ class GetDocumentReply : public DocumentAcceptedReply {
      *
      * @param document The document requested.
      */
-    GetDocumentReply(document::Document::SP document);
-
-    /**
-     * Returns the document retrieved.
-     *
-     * @return The document.
-     */
-    document::Document::SP getDocument();
+    GetDocumentReply(DocumentSP document);
 
     /**
      * Returns the document retrieved.
      *
      * @return The document.
      */
-    std::shared_ptr<const document::Document> getDocument() const;
+    const DocumentSP & getDocument() const { return _document; }
```

Comment:
Since the returned `Document` pointer is no longer const, we have to be careful that callers do the Right Thing™ and don't try to mutate things they shouldn't. Ideally we'd just return a `const document::Document&` here, if possible. Or perhaps split into two accessors, one for by-ref doc and one for by-value shared_ptr<const doc>.

This also applies to the other similar changes in this diff.